script.js

```diff
@@ -0,0 +1,364 @@
+    }
+  }
+
+  function evaluateExpression(first, operator, second) {
+    // Handle a complete expression
+    if (first.saved && operator.saved && second.saved) {
+      // Handle a single minus sign as a second number, like 25x-
+      if (second.value === "-") {
+        return;
+      }
+      result = operate(first.value, operator.value, second.value);
+    }
+    // Handle a single first number followed by a sign
+    if (first.saved && operator.saved && second.unsaved) {
+      second.value = first.value;
+      result = operate(first.value, operator.value, second.value);
+    }
+    // Handle a single first number
+    if (first.saved && operator.unsaved && second.unsaved) {
+      result = first.value;
+    }
+    // Handle an empty expression
+    if (first.unsaved && operator.unsaved && second.unsaved) {
+      result = "0";
+    }
+    // Allow to chain expression when adding a sign
+    if (result !== "Error") {
+      first.value = result;
+      allowNewNumber = true;
+      [operator, second].forEach(element => resetData(element));
+    } else {
+      [first, operator, second].forEach(element => resetData(element));
+    }
+  }
+
+  function operate(first, sign, second) {
+    const a = Number(first);
+    const b = Number(second);
+    switch (sign) {
+      case "+":
+        return String(add(a, b));
+      case "-":
+        return String(subtract(a, b));
+      case "×":
+        return String(multiply(a, b));
+      case "^":
+        return String(power(a, b));
+      case "÷":
+        // Handle a division by zero
+        if (b === 0) {
+          return "Error";
+        }
+        return String(divide(a, b));
+    }
+  }
+
+  function add(a, b) {
+    return a + b;
+  }
+
+  function subtract(a, b) {
+    return a - b;
+  }
+
+  function multiply(a, b) {
+    return a * b;
+  }
+
+  function divide(a, b) {
+    return a / b;
+  }
+
+  function power(a, b) {
+    return a ** b;
+  }
+
+  function addDecimal(pointEvent) {
+    // When user start with "."
+    if (operator.unsaved && first.unsaved && first.integer) {
+      first.value = "0" + pointEvent;
+      saveData(first, true);
+    }
+    // Handle negative number
+    if (operator.unsaved && first.saved && first.integer) {
+      if (first.value.startsWith("-")) {
+        first.value += "0" + pointEvent;
+        first.integer = false;
+      } else {
+        first.value += pointEvent;
+        first.integer = false;
+      }
+    }
+    if (operator.saved && second.unsaved && second.integer) {
+      second.value = "0" + pointEvent;
+      saveData(second, true);
+    }
+    if (operator.saved && second.saved && second.integer) {
+      if (second.value.startsWith("-")) {
+        second.value += "0" + pointEvent;
+        second.integer = false;
+      } else {
+        second.value += pointEvent;
+        second.integer = false;
+      }
+    }
+  }
+
+  function getOpposite() {
+    // When a number consist only in a minus saved we want nothing to happen
```

Comment:
Again here instead of using comments, it'd be better to break this function down into separate functions with descriptive names